ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/PendingTransactions.java

```diff
@@ -204,34 +204,31 @@ private void incrementTransactionRemovedCounter(
   }
 
   public void selectTransactions(final TransactionSelector selector) {
-    synchronized (prioritizedTransactions) {
```

Comment:
There are two other sync locks on `prioritizedTransactions`, should we drop those too?  I din't see adds and remove conflicting, the sync is more essential for the select as it is iterative.

Also, since it's `weaklyConsistant` we may miss removes.  but missing adds is OK.  As an alternative should we keep the sync locks in `selectTransactions` and `addTransaction` but remove the one in `doRemoveTransaction`?

Reply:
The main idea of this PR is to no longer impact the addition or deletion because the evaluation of transactions would be too long. If I keep the synchronization between the `selectTransaction` and the `addTransaction` the problem will still be there

On the other hand I wanted to keep the synchronization between the `addTransaction` and `doRemoveTransaction` because there are parts that can cause some edge cases if I delete it IMO. 

For example I will have to sync this block https://github.com/hyperledger/besu/blob/9062140452fb054f2487a9096395fee7132a6a0b/ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/PendingTransactions.java#L259

And I will also have to synchronize this `if`block 
https://github.com/hyperledger/besu/blob/9062140452fb054f2487a9096395fee7132a6a0b/ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/PendingTransactions.java#L298

So I'm not sure it will change the result much 🤔

Reply:
Yea, we would need to look into what other data structures are accessed and see if they are accessed in the select call.  threading bugs are tricky to test for as they can survive months of testing without manifesting.  Hence the 'sync it all' paranoia of the current class.